tests/libvmi/vmi.go

```diff
@@ -251,11 +251,11 @@ func WithPasstInterfaceWithPort() Option {
 	return WithInterface(InterfaceDeviceWithPasstBinding([]v1.Port{{Port: 1234, Protocol: "TCP"}}...))
 }
 
-func WithNodeAffinityFor(node *k8sv1.Node) Option {
+func WithNodeAffinityFor(nodeName string) Option {
```

Comment:
this change of signature is not self-explanatory. Why is it better with `nodeName`?

I have a feeling that your code change could be less "disruptive" if you add `libnode.GetNodeByName()` to be used in few places.

Reply:
Per what I see, it was changed to support sceanarios where only the name is available (e.g. from the pod) and not the node object.
Fetching the node object from the API at callers seems an overkill.

I do not see a problem with this change, unless in the future other node fields will be used for the affinity. But the future is not clear in this case, so I think we can keep it as is.

Reply:
No, please keep the short name. I just wish I saw the motivation declared in a separate commit.

Reply:
I will separate the first commit into 3 as @xpivarc suggested. Thanks

Reply:
@xpivarc I don't quite follow you most recent comment. I'd love seeing a commit telling me about the signature change and what it's good for.